bin/node-template/node/src/service.rs

```diff
@@ -91,7 +92,7 @@ pub fn new_full(config: Configuration) -> Result<TaskManager, ServiceError> {
 		GrandpaFinalityProofProvider::new_for_service(backend.clone(), client.clone());
 
 	let (network, network_status_sinks, system_rpc_tx, network_starter) =
-		sc_service::build_network(sc_service::BuildNetworkParams {
+		sc_service::build_network::<_, _, _, _, Multihash>(sc_service::BuildNetworkParams {
```

Comment:
Ideally, I think we want `Multihash` to be a default field here, so we don't have to do this whole `_, _, _,` thing. As functions can't have generic fields, but structs can, we could move `build_network` to be a method of `BuildNetworkParams` instead. This would look uglier and be less consistent, but it would atleast solve this problem.

Reply:
Not sure if `BuildNetworkParams` should have a pub field of type `PhantomData`. I made `build_network` a method that only takes one generic type parameter, the `Multihash`.